tests/python/pants_test/engine/test_rules.py

```diff
@@ -212,12 +212,15 @@ def a_from_b_noop(b: B) -> A:
     with self.assertRaises(Exception) as cm:
       create_scheduler(rules)
 
-    self.assert_equal_with_printing(dedent("""
-                     Rules with errors: 1
-                       (A, [B], a_from_b_noop()):
-                         No rule was available to compute B with parameter type SubA
-                     """).strip(),
-                                    str(cm.exception))
+    self.assert_equal_with_printing(
+      dedent(
+        """\
+        Rules with errors: 1
+          Rule(func=a_from_b_noop(), product=A, params=[B]):
```

Comment:
Sorry for bikeshedding, and definitely ignore if you don't care! How would you feel about fully-qualifying the function name? i.e. `pants.python.rules.fmt()` instead of `fmt()`?